daml-foundations/daml-ghc/daml-prim-src/GHC/Num.daml

```diff
@@ -58,37 +58,38 @@ class Additive a  where
     {-# INLINE negate #-}
     negate x            = aunit - x
 
--- | Basic multiplicative class.
--- Instances are expected to respect the following laws:
---
--- **Associativity of (*)**:`(x * y) * z` = `x * (y * z)`
---
--- **Commutativity of (*)**: `x * y` = `y * x`
+-- | Use the `Multiplicative` class for types that can be multiplied.
+-- Instances have to follow these rules:
 --
--- `x * one` = `x`
+-- * (*) is associative, ie:`(x * y) * z` = `x * (y * z)`
+-- * (*) is commutative, ie: `x * y` = `y * x`
+-- * `x * one` = `x`
 --
 class Multiplicative a where
+    -- | Multipy the arguments together
     (*) : a -> a -> a
+    -- | The multiplicative identity for the type. For example, for numbers, this is 1.
     munit : a
     -- | `x ^ n` raises `x` to the power of `n`.
     (^): a -> Int -> a
 
--- | Basic numeric class.
--- Instances are usually expected to respect the following law
--- (in addition to laws from being additive and multiplicative)
+-- | `Number` is a class for basic numerical types.
+-- As well as the rules for `Additive` and `Multiplicative`, instances
+-- also have to follow these rules:
 --
--- **Distributivity of (*) with respect to (+)**:
+-- * (*) is distributive with respect to (+), ie:
 --     `a * (b + c)` = `(a * b) + (a * c)` and `(b + c) * a` = `(b * a) + (c * a)`
 class (Additive a, Multiplicative a) => Number a where
   -- no methods
 
+-- | The `Signed` is for the sign of a number.
 class Signed a where
```

Comment:
@neil-da Should we add a super class constraint `Number a` to this? What else does have a sign?